server/shared/publishing/helpers.js

```diff
@@ -56,6 +56,18 @@ function updateRepositoryCatalog(repository, publishedAt) {
   });
 }
 
+function deprecateRepository(repository, deprecatedAt) {
```

Comment:
I've separated concerns of repo deprecation and repo update with this method, even though we could probably refactor "updateRepositoryCatalog" method to deprecate repos under certain conditions but I'm not sure we should.